app/lib/activitypub/activity/create.rb

```diff
@@ -340,15 +343,23 @@ def fetch_replies(status)
     ActivityPub::FetchRepliesWorker.perform_async(status.id, uri) unless uri.nil?
   end
 
-  def conversation_from_uri(uri)
-    return nil if uri.nil?
-    return Conversation.find_by(id: OStatus::TagManager.instance.unique_tag_to_local_id(uri, 'Conversation')) if OStatus::TagManager.instance.local_id?(uri)
+  def conversation_from_context
+    return if @object['context'].nil?
+
+    uri          = value_or_id(@object['context'])
+    conversation = ActivityPub::TagManager.instance.uri_to_resource(uri, Conversation)
+
+    return conversation if conversation.present?
 
-    begin
-      Conversation.find_or_create_by!(uri: uri)
-    rescue ActiveRecord::RecordInvalid, ActiveRecord::RecordNotUnique
-      retry
+    conversation_json = begin
+      if @object['context'].is_a?(Hash)
+        @object['context']
+      else
+        fetch_resource(uri)
+      end
     end
+
+    Conversation.create!(uri: uri, inbox_url: conversation_json['inbox'])
```

Comment:
Is the conversation's inbox ever used? What happens if the object doesn't have one? [`context` attribute](https://www.w3.org/TR/activitystreams-vocabulary/#dfn-context) is very ill-defined in ActivityStreams :/

Reply:
In ActivityPub::DistributionWorker, the `deliver_to_parent!` method uses that inbox.